core/src/main/java/org/apache/spark/util/ChildFirstURLClassLoader.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+package org.apache.spark.util;
+
+import sun.misc.CompoundEnumeration;
+
+import java.io.IOException;
+import java.net.URL;
+import java.util.Enumeration;
+
+/**
+ * A mutable class loader that gives preference to its own URLs over the parent class loader
+ * when loading classes and resources.
+ */
+public class ChildFirstURLClassLoader extends MutableURLClassLoader {
+
+  {
```

Comment:
This is an initializer blocks for instance variables:
https://docs.oracle.com/javase/tutorial/java/javaOO/initial.html

It is called for each instance.

Based on the PR description: `required to register themselves at their class initialization time by invoking the ClassLoader.registerAsParallelCapable method` I guess you meant to use a static block.

(If still an instance block is needed, I think for one constructor it is better just to move the logic right after the calling `super`.)

Reply:
My bad, my intention was to keep it in static block.. i have corrected it.